### DISCUSSION TITLE
Run script on test network

### DISCUSSION THREAD

**Discussion by CaledoniaProject** [foundry-rs/foundry#2457] [Help]
I'm using `--fork-url` to test the contract without actually modifying the state on chain. 

How can I execute the contract code on the real chain? I also want to print the transaction hash as well.

**Comment by onbjerg** [comment]
Use the `vm.broadcast`/`vm.startBroadcast`/`vm.stopBroadcast` cheatcodes to mark what transactions you want to submit on-chain. Then use the `--broadcast` flag in conjunction with your preferred signer method. Refer to the docs [here](https://book.getfoundry.sh/tutorials/solidity-scripting) and [here](https://book.getfoundry.sh/reference/forge/forge-script)
